extensions/positron-notebook-controllers/src/statusBar.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import { NotebookControllerManager } from './notebookControllerManager';
+import { formatCount } from './utils';
+
+export const SHOW_EXECUTION_INFO_SECTION = 'notebook.experimental.showExecutionInfo';
+
+export function registerExecutionInfoStatusBar(
+	disposables: vscode.Disposable[], manager: NotebookControllerManager
+): void {
+	const item = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right);
+
+	/** Update the item's visibility based on user configuration. */
+	const updateVisibility = () => {
+		const showExecutionInfo = vscode.workspace.getConfiguration().get<boolean>(SHOW_EXECUTION_INFO_SECTION);
+		if (showExecutionInfo) {
+			item.show();
+		} else {
+			item.hide();
+		}
+	};
+
+	disposables.push(item);
+
+	// Update the item's text when an execution starts.
+	disposables.push(manager.onDidStartExecution((e) => {
+		item.text = `Executing ${formatCount(e.cells.length, 'cell')}`;
+	}));
+
+	// Update the item's text when an execution ends.
+	disposables.push(manager.onDidEndExecution((e) => {
+		const durationSeconds = Math.round(e.duration / 1000);
+		item.text = `Executed ${formatCount(e.cells.length, 'cell')} ` +
+			`in ${formatCount(durationSeconds, 'second')}`;
```

Comment:
Will we potentially want more precision than the second level?

Reply:
This is how they format time stamps. Not sure how fancy we need to get for testing though. Maybe @testlabauto  has an opinion?
https://github.com/posit-dev/positron/blob/25f4eaedb24cc555ff60683beffe465c32ae968a/src/vs/workbench/contrib/notebook/browser/contrib/cellStatusBar/executionStatusBarItemController.ts#L23-L37

Reply:
We may want these sorts of refinements/additions in future. I opted for something super simple for now that matches Jupyter Lab's behavior enough to satisfy @testlabauto's benchmarking needs.